File: native/iceberg/oracle/src/main/java/IcebergTableOracle.java
import java.nio.file.*;
import com.fasterxml.jackson.databind.*;
import com.fasterxml.jackson.databind.node.*;
import org.apache.iceberg.TableMetadata;
import org.apache.iceberg.TableMetadataParser;
public final class IcebergTableOracle {
 public static void main(String[] args) throws Exception {
  Path base=Path.of(args[0]);ObjectMapper mapper=new ObjectMapper();ArrayNode results=mapper.createArrayNode();
  for(JsonNode c:mapper.readTree(base.resolve("results.json").toFile()).get("results")){
   String id=c.get("id").asText(),raw=Files.readString(Path.of(c.get("path").asText()));ObjectNode r=mapper.createObjectNode().put("id",id).put("umf",c.get("status").asText());TableMetadata original;
   try{original=TableMetadataParser.fromJson(raw);r.put("nativeAccepted",true);}catch(Exception e){r.put("nativeAccepted",false).put("nativeError",e.getClass().getSimpleName()+": "+e.getMessage());results.add(r);continue;}
   if(c.get("status").asText().equals("roundtripped")){
    JsonNode expected=mapper.readTree(TableMetadataParser.toJson(original));
    for(String f:new String[]{"json","yaml"}){TableMetadata actual=TableMetadataParser.fromJson(Files.readString(base.resolve(id+"."+f+".json")));if(!expected.equals(mapper.readTree(TableMetadataParser.toJson(actual))))throw new IllegalStateException("Round trip differs: "+id);}
    TableMetadata edited=TableMetadataParser.fromJson(Files.readString(base.resolve(id+".edited.json")));((ObjectNode)expected).put("location","s3://umf-fixture/relocated/"+id);if(!expected.equals(mapper.readTree(TableMetadataParser.toJson(edited))))throw new IllegalStateException("Unexpected edit: "+id);r.put("formatsAgree",2).put("editVerified",true);
   }results.add(r);
  }
  ObjectNode report=mapper.createObjectNode().put("runtime","Apache Iceberg Java 1.11.0").put("javaVersion",System.getProperty("java.version")).set("results",results);mapper.writerWithDefaultPrettyPrinter().writeValue(base.resolve("java-oracle-results.json").toFile(),report);System.out.println("Native Java tables checked: "+results.size());
 }
}

File: scripts/iceberg-table-oracle.py
import json
from pathlib import Path
from pyiceberg.table.metadata import TableMetadataUtil
base=Path('fixtures/iceberg/table');results=[]
def native(text):
 try:model=TableMetadataUtil.parse_raw(text)
 except Exception as e:return {'parseAccepted':False,'exportAccepted':False,'output':None,'error':str(e)}
 try:return {'parseAccepted':True,'exportAccepted':True,'output':model.model_dump_json(by_alias=True),'error':None}
 except Exception as e:return {'parseAccepted':True,'exportAccepted':False,'output':None,'error':str(e)}
def comparable(r):return (r['parseAccepted'],r['exportAccepted'],r['output'])
for c in json.load(open(base/'results.json'))['results']:
 raw=Path(c['path']).read_text();expected=native(raw);r={'id':c['id'],'umf':c['status'],'nativeAccepted':expected['parseAccepted'],'nativeExported':expected['exportAccepted'],'nativeError':expected['error']}
 if c['status']=='roundtripped':
  for f in ['json','yaml']:
   text=(base/(c['id']+'.'+f+'.json')).read_text();assert json.loads(text)==json.loads(raw);assert comparable(native(text))==comparable(expected),c['id']
  r['formatsAgree']=2;r['nativeReemissionCompared']=expected['exportAccepted'];edited=(base/(c['id']+'.edited.json')).read_text();input=json.loads(raw);input['location']='s3://umf-fixture/relocated/'+c['id'];assert json.loads(edited)==input;actual=native(edited);assert comparable(actual)==comparable(native(json.dumps(input)));r['editNativeParsed']=actual['parseAccepted'];r['editNativeExported']=actual['exportAccepted']
 results.append(r)
(base/'oracle-results.json').write_text(json.dumps({'runtime':'PyIceberg 0.11.0','results':results},indent=2)+'\n');print({'files':len(results),'nativeParsed':sum(r['nativeAccepted'] for r in results),'nativeExported':sum(r['nativeExported'] for r in results),'roundtrips':sum(r.get('formatsAgree',0) for r in results)})
